### PR TITLE
Cherry pick 2680 to 6x

### PR DESCRIPTION
**Pull request by keegan-caruso**
Cherry pick #2680 to 6x with modifications to handle differences in JsonClaimSet

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: src/Microsoft.IdentityModel.Validators/AadTokenValidationParametersExtension.cs [hunk 1]
```diff
@@ -67,9 +79,16 @@ internal static bool ValidateIssuerSigningKey(SecurityKey securityKey, SecurityT
                 if (string.IsNullOrWhiteSpace(signingKeyIssuer))
                     return true;
 
-                var tenantIdFromToken = AadIssuerValidator.GetTenantIdFromToken(securityToken);
+                var tenantIdFromToken = GetTid(securityToken);
                 if (string.IsNullOrEmpty(tenantIdFromToken))
-                    return true;
+                {
+#if !NET45
+                    if (DontFailOnMissingTid())
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Validators/AadTokenValidationParametersExtension.cs:86]: don't app context switches exist in .NET45?

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @keegan-caruso

**Review by jmprieur — APPROVED** [review]

File: src/Microsoft.IdentityModel.Validators/AadTokenValidationParametersExtension.cs (hunk 1 above)
Comment (keegan-caruso) [reply src/Microsoft.IdentityModel.Validators/AadTokenValidationParametersExtension.cs:86]: They were added in 4.6

https://learn.microsoft.com/en-us/dotnet/api/system.appcontext?view=netframework-4.6

**Pull request merged**